### ISSUE TITLE
No native client found (Android)

### ISSUE THREAD

**Issue by preko96** [bugsnag/bugsnag-react-native#274]
### Description
After linking Bugsnag to a React Native project (using react-native link), the bugsnag module is not found on the android side.

### Issue
The linking on the android side.

IMPORTANT NOTICE:
I am using react-native-navigation where I extend my MainApplication from the NavigationApplication class instead of the ReactApplication

### Environment

Library versions:

```shell
bugsnag-react-native@2.10.3
react-native@0.55.4
```

- iOS/Android version(s):
NEXUS_5X_API_28_86x

- simulator/emulator or physical device?:
simulator

- debug mode or production?:
DEBUG

- [ ] (Android only) `BugsnagReactNative.start(this)` is present in the
  `onCreate` method of your `MainApplication` class?

YES

### Example code snippet

```js
import { Client, Configuration } from 'bugsnag-react-native';
const config = new Configuration(Config.BUGSNAG_API_KEY);
config.appVersion = require('./package.json').version;
const bugsnag = new Client(config);
```

<details><summary>Error messages: Bugsnag: No native client found. Is Bugsnag ReactNative installed in your native code project?</summary>

</details>

**Comment by preko96** [comment]
Well, I might wrote the bug report too fast.

After going to Build-Rebuild (Android Studio) everything seems to work.
But one thing is quite strange.

If I add 

```js
@Override
    public void onCreate() {
        super.onCreate();
        BugsnagReactNative.start(this);
    }
```
Its crashing my app immediately. (the MainApplication extends from NavigationApplication, AND the NavigationApplication extends Application and implements ReactApplication, so overriding should do the job...)

But even so, it's seems to work even without  
```js  
BugsnagReactNative.start(this);
``` 
Is it alright this way?

**Comment by fractalwrench** [comment]
`BugsnagReactNative.start(this);` is required for things to work properly on Android. Would you mind sharing a stacktrace of how it's crashing your app?

**Comment by preko96** [comment]
Of course

```sta

**Comment by preko96** [comment]
After adding this line, it's seems to work properly.

```xml
<meta-data android:name="com.bugsnag.android.API_KEY" android:value="*** Your bugsnag API key ***"/>
```

**Comment by eightyfive** [comment]
Thanks @preko96 

This should be written somewhere in the docs.

**Comment by mattdyoung** [comment]
It is documented:
https://docs.bugsnag.com/platforms/react-native/react-native/enhanced-native-integration/#android-native-installation

But the fact you didn't find it shows that maybe it's not prominent enough.

We're in the process of making a number of improvements to simplify configuration of the library so we'll ensure this is made clearer in the future if it's still required.

**Comment by eightyfive** [comment]
@mattdyoung To be fair, I rushed a bit my implementation so I might have missed this piece of documentation.

I find Bugsnag documentation overall very decent.

Some reasons I struggled a bit:
- I use `react-native-navigation` in my project, hence my `MainApplication.java` does not extend the standard React class
- I've been rushing/tweaking my `MainApplication.java` class based on that [source code](https://github.com/bugsnag/bugsnag-react-native/blob/master/examples/codepush/android/app/src/main/java/com/bugsnagreactnativeexample/MainApplication.java), not on proper documentation
- Finally I googled my way to here

There is a piece of documentation that helped me a lot, it was showing different `MainApplication.java` implementations depending on if you are extending the standard React class or not, but I cannot find it anymore. Maybe it was a SO answer.

This was valuable: Different `MainApplication.java` "manual linking" depending on what you use.
